packages/aws-cdk-lib/aws-codepipeline-actions/lib/ec2/deploy-action.ts

```diff
@@ -0,0 +1,232 @@
+   *
+   * You must have already created, tagged, and installed the SSM agent on all instances.
+   *
+   * @default Ec2InstanceType.EC2
+   */
+  readonly instanceType?: Ec2InstanceType;
+
+  /**
+   * The location of the target directory you want to deploy to.
+   * Use an absolute path like `/home/ec2-user/deploy`.
+   */
+  readonly targetDirectory: string;
+
+  /**
+   * The number or percentage of instances that can deploy in parallel.
+   *
+   * Valid range:
+   * - targets: from `'1'` to number of your instances
+   * - percentage: from `'1%'` to `'99%'`
+   *
+   * @default - No configuration
+   */
+  readonly maxBatch?: Ec2MaxInstances;
+
+  /**
+   * Stop the task after the task fails on the specified number or percentage of instances.
+   *
+   * Valid range:
+   * - targets: from `'1'` to number of your instances
+   * - percentage: from `'1%'` to `'99%'`
+   *
+   * @default - No configuration
+   */
+  readonly maxError?: Ec2MaxInstances;
+
+  /**
+   * The list of target groups for deployment. You must have already created the target groups.
+   *
+   * Target groups provide a set of instances to process specific requests.
+   * If the target group is specified, instances will be removed from the target group before deployment and added back to the target group after deployment.
+   *
+   * @default - No target groups
+   */
+  readonly targetGroups?: elbv2.ITargetGroup[];
+
+  /**
+   * Path to the executable script file that runs BEFORE the Deploy phase.
+   * It should start from the root directory of your uploaded source artifact.
+   * Use an absolute path like `uploadDir/preScript.sh`.
+   *
+   * @default - No script
+   */
+  readonly preScript?: string;
+
+  /**
+   * Path to the executable script file that runs AFTER the Deploy phase.
+   * It should start from the root directory of your uploaded source artifact.
+   * Use an absolute path like `uploadDir/postScript.sh`.
+   */
+  readonly postScript: string;
+}
+
+/**
+ * The type of instances or SSM nodes created in Amazon EC2.
+ * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-EC2Deploy.html#action-reference-EC2Deploy-parameters
+ */
+export enum Ec2InstanceType {
+  /** Amazon EC2 instances */
+  EC2 = 'EC2',
+  /** AWS System Manager (SSM) managed nodes */
+  SSM_MANAGED_NODE = 'SSM_MANAGED_NODE',
+}
+
+/**
+ * Number or percentage of max instances for EC2 deploy action.
+ */
+export abstract class Ec2MaxInstances {
+  /**
+   * Max number of instances.
+   *
+   * Valid range: from 1 to number of your instances
+   */
+  public static targets(targets: number): Ec2MaxInstances {
+    if (!Token.isUnresolved(targets) && !(targets >= 1 && Number.isInteger(targets))) {
+      throw new UnscopedValidationError(`targets must be a positive integer. got ${targets}`);
+    }
+    return { value: Tokenization.stringifyNumber(targets) };
+  }
+
+  /**
+   * Max percentage of instances.
+   *
+   * Valid range: from 1 to 99
```

Comment:
Validation based on AWS console:
![image](https://github.com/user-attachments/assets/e219102e-f548-4ef5-8f05-32f855ed7560)

Reply:
I've succeeded to deploy with `maxBatch: '100%'`. The note of the AWS console seems inaccurate.

Reply:
I've updated max allowed percentage to 100%.